### ISSUE TITLE
Death Prophet's Exorcism does not heal full amount

### ISSUE THREAD

**Issue by deronek** [ValveSoftware/Dota2-Gameplay#2923]
### Ability name

Exorcism

### Description

- **Match ID**: 6747919807
- **Clear description of the bug**: During big teamfight, Death Prophet used Exorcism, which was dealing lots of damage for the whole fight (estimated around ~3600 damage). At the end of the fight, Death Prophet was healed for 105 HP, while the heal should be around 900 HP instead.
- **Timestamp when the bug occurs**: Exorcism started at 34:56, finished at 35:30
- **Steps to reproduce this bug**: Unknown, happened in the game and could not be reproduced normally
- **Images and videos showing the bug:**
![unknown](https://user-images.githubusercontent.com/62315827/189238816-4b4cacef-9744-4b97-ae87-a815e98605c3.png)
Combat log showing the amount of heal, significantly less than dealt damage.

### Example Match ID (and possibly Timestamp)

6747919807 - from 34:56 to 35:30

**Comment by TheUnholyWrath** [comment]
It was the teleport. If you move more than 2000 distance away from your spirits they will despawn and new ones will respawn. Seems like an unintentional side effect of an intended mechanic.

**Comment by deronek** [comment]
> It was the teleport. If you move more than 2000 distance away from your spirits they will despawn and new ones will respawn. Seems like an unintentional side effect of an intended mechanic.

Makes sense, thanks for the clarification. I leave it up for the developers if it's 100 % intended this way.

**Comment by dobacetr** [comment]
I don't think despawn-respawn mechanics should have this kind of effect. The same spirit should be respawning, with the same damage dealt accumulated. I would consider this a bug.
